Automattic/vip
# [PIE-3548] Removes `publish-please` and replaces it with simple custom script

## Description

Jira: https://vipjira.atlassian.net/browse/PIE-3548

NPM will give some `deprecated` warnings when installing the CLI:

```
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated source-map-url@0.4.0: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated cross-spawn-async@2.2.5: cross-spawn no longer requires a build toolchain, use it instead
```

Those are all related to `publish-please`:

```
npm why cross-spawn-async
cross-spawn-async@2.2.5 dev
node_modules/cross-spawn-async
  cross-spawn-async@"^2.1.6" from cp-sugar@1.0.0
  node_modules/cp-sugar
    cp-sugar@"1.0.0" from publish-please@5.5.2
    node_modules/publish-please
      dev publish-please@"5.5.2" from the root project 
```

and we're already at its latest version. There have been no activity on that [repo](https://github.com/inikulin/publish-please) since 2020. 

I've put together a crude node script that accomplishes the basic functionality `publish-please` would give us. We can add more if neeeded.

## Steps to Test

1. Check out PR.
1. run `npm install`
1. Try the script with `npm publish --dry-run`. Test the different rules:

- Try running that command on a branch other than `trunk` or change the allowed branch`
- Try publishing with local changes
- Change to a Node version less that 16.16 and try publishing
- Make tests fail and try publishing
